ang/civicase/CaseDetailsPeopleTab.js

```diff
@@ -79,11 +79,24 @@
       if (onlyChecked || $scope[tab + 'SelectionMode'] === 'checked') {
         return _.collect(_.filter($scope[tab], {checked: true}), idField);
       } else if ($scope[tab + 'SelectionMode'] === 'all') {
-        return _.collect($scope[tab], idField);
+        return _.collect(_.filter($scope[tab], function (el) {
+          return el.contact_id;
+        }), idField);
       }
       return [];
     };
 
+    /**
+     * Filter the records on basis of contact_role key
+     *
+     * @param {Array} records
+     */
+    $scope.getRecordCountWithContacts = function (records) {
+      return _.filter(records, function (record) {
+        return record.contact_id;
```

Comment:
what is a `record` that might contain a `contact_id` property? What does it mean when it doesn't have the property?

Reply:
As per my above comment. There may be roles which are not assigned to a contact. So we need to filer down the roles which have some contact assigned to them.